MSAL/test/app/MSALStressTestHelper.m

```diff
@@ -0,0 +1,200 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALStressTestHelper.h"
+#import "MSALTestAppTelemetryViewController.h"
+#import "MSALTestAppSettings.h"
+#import "NSURL+MSALExtensions.h"
+#import "MSALAuthority.h"
+#import "MSALAccessTokenCacheItem+TestAppUtil.h"
+
+@implementation MSALStressTestHelper
+
+static dispatch_semaphore_t s_sem = nil;
+static BOOL s_stop = NO;
+static BOOL s_runningTest = NO;
+
+#pragma mark - Helpers
+
++ (void)expireAllTokens
+{
+    MSALKeychainTokenCache *cache = MSALKeychainTokenCache.defaultKeychainCache;
+    NSArray *tokenCacheItems = [cache getAccessTokenItemsWithKey:nil context:nil error:nil];
+    
+    for (MSALAccessTokenCacheItem *item in tokenCacheItems)
+    {
+        item.expiresOnString = [NSString stringWithFormat:@"%qu", (uint64_t)[[NSDate dateWithTimeIntervalSinceNow:-1.0] timeIntervalSince1970]];
+        [cache addOrUpdateAccessTokenItem:item context:nil error:nil];
+    }
+}
+
+#pragma mark - Stress tests
+
++ (void)testAcquireTokenSilentWithExpiringToken:(BOOL)expireToken
+                               useMultipleUsers:(BOOL)multipleUsers
+                                    application:(MSALPublicClientApplication *)application
+{
+    MSALTestAppSettings *settings = [MSALTestAppSettings settings];
+    NSArray<MSALUser *> *users = [application users:nil];
+    
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        
+        s_sem = dispatch_semaphore_create(10);
```

Comment:
static semaphores should only be created once, ideally in a dispatch_once, or in +initialize